pkg/bpf/bpf.go

```diff
@@ -305,6 +314,10 @@ func OpenOrCreateMap(path string, mapType int, keySize, valueSize, maxEntries ui
 		if err != nil {
 			return 0, isNewMap, err
 		}
+
+		// Need to close this fd since the next ObjGet
+		// returns a different fd.
+		ObjClose(fd)
```

Comment:
```
  if _, err = os.Stat(path); os.IsNotExist(err) {
    [...]
    err = ObjPin(fd, path)
    if err != nil {
        ObjClose(fd)
        return 0, isNewMap, err
    }

    return fd, isNewMap, err
  }
```

No, but similar to above I mean. The ObjGet() is for the case when a node exists in the fs, so we try to fetch that one and later check fdinfo, etc.

When we have to go and create the map, and then ObjPin() it, we already have the map fd, and it still stays around after pinning. You can pin the same fd as often as you want. Thus, after pinning, we can just return it to the caller. I was just trying to say that the extra ObjClose() and ObjGet() detour is more unnecessary and error prone (as in can fail again for various reasons) than just returning the fd from there.